pxtblocks/blocklyloader.ts

```diff
@@ -913,21 +1040,168 @@ namespace pxt.blocks {
+            'variables_set',
+            lf("assign the value of a variable"),
+            lf("Sets this variable to be equal to the input."),
+            '/blocks/variables/assign'
+        );
+
+        // pxt variables_change
         Blockly.Blocks['variables_change'] = {
-            init: function () {
-                this.appendDummyInput()
-                    .appendField("change")
-                    .appendField(new Blockly.FieldVariable("item"), "VAR");
-                this.appendValueInput("VALUE")
-                    .setCheck("Number")
-                    .appendField("by");
-                this.setInputsInline(true);
-                this.setPreviousStatement(true);
-                this.setNextStatement(true);
-                this.setTooltip(lf("Changes the value of the variable by this amount"));
-                this.setHelpUrl('/reference/assign');
-                this.setColour(blockColors['variables']);
-            }
-        };
+	    init: function () {
+		this.jsonInit({
+		  "message0": lf("change %1 by %2"),
+		  "args0": [
+		    {
+		      "type": "field_variable",
+		      "name": "VAR",
+		      "variable": "item"
+		    },
+		    {
+		      "type": "input_value",
+		      "name": "VALUE",
+		      "check": "Number"
+		    }
+		  ],
+		  "inputsInline": true,
+		  "previousStatement": null,
+		  "nextStatement": null,
+		  "colour": blockColors['variables']
+		});
+
+		setHelpResources(this,
+		    'variables_change',
+		    lf("update the value of a number variable"),
+		    lf("Changes the value of the variable by this amount"),
+		    '/blocks/variables/change-var'
+		);
+	    }
+	};
+    }
+
+    function initLogic() {
+        let msg:any = Blockly.Msg;
+
+        // builtin controls_if
+        msg.CONTROLS_IF_MSG_IF = lf("_logic_if");
+        msg.CONTROLS_IF_MSG_THEN = lf("_logic_then");
+        msg.CONTROLS_IF_MSG_ELSE = lf("_logic_else");
+        msg.CONTROLS_IF_MSG_ELSEIF = lf("_logic_elseif");
+        msg.CONTROLS_IF_TOOLTIP_1 = lf("If a value is true, then do some statements.");
+        msg.CONTROLS_IF_TOOLTIP_2 = lf("If a value is true, then do the first block of statements. Otherwise, do the second block of statements.");
+        msg.CONTROLS_IF_TOOLTIP_3 = lf("If the first value is true, then do the first block of statements. Otherwise, if the second value is true, do the second block of statements.");
+        msg.CONTROLS_IF_TOOLTIP_4 = lf("If the first value is true, then do the first block of statements. Otherwise, if the second value is true, do the second block of statements. If none of the values are true, do the last block of statements.");
+        installHelpResources(
+            'controls_if',
+            lf("a conditional statement"),
+            undefined,
+            "blocks/logic/if"
+        );
+
+        // builtin logic_compare
+	Blockly.Blocks['logic_compare'].init = function() {
+	    let OPERATORS = [
+		[lf("_op_eq"),  'EQ'],  // =
```

Comment:
I added a commit to use `==`, `&&` and `||` instead of `_op_`?? strings.

How do you think about other `_op_`?? strings like `_op_add` and `_op_sub` etc? I think these strings *may* be better than bare symbols like `+` or `-`. However, I don't insist on this.